### PR TITLE
DrugInfo & RaxaDrug controllers and resources

### PR DESCRIPTION
**Pull request by diegodayan**
Functionality in this pull request includes:
- DrugInfo POJO, service, DAO, resource and controller. Includes attributes such as price, cost, name, description.
- RaxaDrug resource and controller (for creating, updating, retrieving OpenMRS Drugs).

### CODE REVIEW

File: omod/src/main/java/org/raxa/module/raxacore/web/v1_0/controller/DrugInfoController.java [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+	 */
+	@RequestMapping(method = RequestMethod.POST)
+	@WSDoc("Save New DrugInfo")
+	@ResponseBody
+	public Object createNewDrugInfo(@RequestBody SimpleObject post, HttpServletRequest request, HttpServletResponse response)
+	        throws ResponseException {
+		initDrugInfoController();
+		DrugInfo drugInfo = new DrugInfo();
+		
+		// TODO: how to retrieve the real Drug from OpenMRS so we check
```
Comment (diegodayan) [inline omod/src/main/java/org/raxa/module/raxacore/web/v1_0/controller/DrugInfoController.java]: @johnstoecker can you please help me with this one?

File: api/src/main/resources/RaxacoreDrugInfo.hbm.xml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+<!DOCTYPE hibernate-mapping PUBLIC
+    "-//Hibernate/Hibernate Mapping DTD 3.0//EN"
+    "http://hibernate.sourceforge.net/hibernate-mapping-3.0.dtd">
+<hibernate-mapping package="org.raxa.module.raxacore">    
+    <class name="DrugInfo" table="raxacore_drug_info">
+        <id name="drugInfoId" column="drug_info_id">
+            <generator class="native"/>
+        </id>
+        <property name="drugId"   type="java.lang.Integer" column="drug_id" insert="false" update="false"/>
+        <many-to-one name="drug"      class="org.openmrs.Drug" column="drug_id" not-null="true" unique="true" />
```
Comment (diegodayan) [inline api/src/main/resources/RaxacoreDrugInfo.hbm.xml:12]: @johnstoecker, one-to-one node wasn't working for me, so I used this one. I understand that because the liquibase file forces a unique index in this column, then there will be no duplicates in the DB. But I don't know if there are any issues that this can produce with Hibernate.

File: omod/src/main/java/org/raxa/module/raxacore/web/v1_0/controller/DrugInfoController.java (hunk 1 above)
Comment (johnstoecker) [inline omod/src/main/java/org/raxa/module/raxacore/web/v1_0/controller/DrugInfoController.java]: Use Context.getConceptService().getDrug(drug_id)

File: api/src/main/resources/RaxacoreDrugInfo.hbm.xml (hunk 2 above)
Comment (johnstoecker) [inline api/src/main/resources/RaxacoreDrugInfo.hbm.xml:12]: Okay, we should use <property name="drug" here with the unique="true" ..../>

Comment (johnstoecker) [inline api/src/main/resources/RaxacoreDrugInfo.hbm.xml:12]: Hm, my comment isn't coming, let me try again.....
< property name="drug" unique="true' />
(instead of many-to-one use property)

Comment (diegodayan) [inline api/src/main/resources/RaxacoreDrugInfo.hbm.xml:12]: It doesn't work if I do it like that. It throws this exception when I run the JUnit tests for saving a DrugInfo object:

ERROR - JDBCExceptionReporter.logExceptions(234) |2012-08-21 10:34:24,753| Value too long for column "DRUG BINARY(255)": "X'aced0005737200106f72672e6f70656e6d72732e44727567000000000000011d0200094c000b636f6d62696e6174696f6e7400134c6a6176612f6c616e672f... (409)"; SQL statement:
insert into raxacore_drug_info (drug_info_id, drug, price, cost, name, description, uuid, creator, date_created, retired, retired_by, date_retired, retire_reason, changed_by, date_changed) values (null, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?) [90005-135]

Comment (johnstoecker) [inline api/src/main/resources/RaxacoreDrugInfo.hbm.xml:12]: Ah, okay. I think to get one-to-one working, we also have to put the <one-to-one...> tag in Drug, but we don't want to do that and modify core OpenMRS. So, for now, keep many-to-one, and add a comment the it should be one-to-one but don't want to modify core drug mapping files.

File: omod/src/main/java/org/raxa/module/raxacore/web/v1_0/controller/DrugInfoController.java (hunk 1 above)
Comment (diegodayan) [inline omod/src/main/java/org/raxa/module/raxacore/web/v1_0/controller/DrugInfoController.java]: Just added in the new commit: https://github.com/diegodayan/raxacore/commit/cfc507bf1051e253370580e6f6bdab18e9208b57

File: api/src/main/resources/RaxacoreDrugInfo.hbm.xml (hunk 2 above)
Comment (diegodayan) [inline api/src/main/resources/RaxacoreDrugInfo.hbm.xml:12]: Ok, thanks!

**Comment by diegodayan** [comment]
Already includes POST for OpenMRS Drugs!

**Comment by johnstoecker** [comment]
Merged at: https://github.com/Raxa/raxacore/pull/33.
Note I changed drugGroupId to id as per discussion with @nathanleiby
